tests/common/snappi_tests/snappi_fixtures.py

```diff
@@ -19,6 +19,12 @@
 from tests.snappi_tests.variables import dut_ip_start, snappi_ip_start, prefix_length, \
     dut_ipv6_start, snappi_ipv6_start, v6_prefix_length, pfcQueueGroupSize, \
     pfcQueueValueDict          # noqa: F401
+from tests.snappi_tests.variables import MULTIDUT_PORT_INFO, MULTIDUT_TESTBED
+from tests.snappi_tests.files.helper import skip_warm_reboot
```

Comment:
We do not permit cross-module dependencies within the repo. This means that in the `common` folder, functions from feature folders like `snappi_tests` should not be imported. Therefore, please relocate the shared functions in `snappi_tests` to a common location.

Reply:
Working on the move to other file. I will be moving the new fixture functions to tests/snappi_tests/files/helper.py. Pls let me know if that is ok. thanks.

Reply:
Sure, please go ahead.

Reply:
@yutongzhang-microsoft : Completed the move. Thanks. The failures below are not related to this change, they are due to unconfigured-RP at that time.

```
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================================================================================================================== PASSES ===========================================================================================================================
___________________________________________________________________________________________ test_pfc_pause_single_lossless_prio[multidut_port_info1-yy39top-lc4|3] ___________________________________________________________________________________________
___________________________________________________________________________________________ test_pfc_pause_single_lossless_prio[multidut_port_info1-yy39top-lc4|4] ___________________________________________________________________________________________
__________________________________________________________________________________________________ test_pfc_pause_multi_lossless_prio[multidut_port_info1] ___________________________________________________________________________________________________
_____________________________________________________________________________________ test_pfc_pause_single_lossless_prio_reboot[multidut_port_info1-cold-yy39top-lc4|3] _____________________________________________________________________________________
_____________________________________________________________________________________ test_pfc_pause_single_lossless_prio_reboot[multidut_port_info1-cold-yy39top-lc4|4] _____________________________________________________________________________________
____________________________________________________________________________________________ test_pfc_pause_multi_lossless_prio_reboot[multidut_port_info1-cold] _____________________________________________________________________________________________
-------------------------------------------------------------------------------- generated xml file: /run_logs/ixia/pfcwd-m2o/2024-10-29-20-07-30/tr_2024-10-29-20-07-30.xml ---------------------------------------------------------------------------------
INFO:root:Can not get Allure report URL. Please check logs
------------------------------------------------------------------------------------------------------------------- live log sessionfinish -------------------------------------------------------------------------------------------------------------------
23:18:54 __init__.pytest_terminal_summary         L0067 INFO   | Can not get Allure report URL. Please check logs
================================================================================================================== short test summary info ===================================================================================================================
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio[multidut_port_info1-yy39top-lc4|3]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio[multidut_port_info1-yy39top-lc4|4]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py::test_pfc_pause_multi_lossless_prio[multidut_port_info1]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio_reboot[multidut_port_info1-cold-yy39top-lc4|3]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio_reboot[multidut_port_info1-cold-yy39top-lc4|4]
PASSED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py::test_pfc_pause_multi_lossless_prio_reboot[multidut_port_info1-cold]
SKIPPED [4] snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py:138: Reboot type warm is not supported on cisco-8000 switches
SKIPPED [4] snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py:138: Reboot type fast is not supported on cisco-8000 switches
SKIPPED [2] snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py:197: Reboot type warm is not supported on cisco-8000 switches
SKIPPED [2] snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py:197: Reboot type fast is not supported on cisco-8000 switches
ERROR snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio[multidut_port_info0-yy39top-lc4|3] - Failed: Processes "['analyze_logs--<MultiAsicSonicHost yy39top-lc4>']" failed with exit code "1"
ERROR snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio[multidut_port_info0-yy39top-lc4|4] - Failed: Processes "['analyze_logs--<MultiAsicSonicHost yy39top-lc4>']" failed with exit code "1"
ERROR snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio[multidut_port_info1-yy39top-lc4|3] - Failed: Processes "['analyze_logs--<MultiAsicSonicHost yy39top-lc4>']" failed with exit code "1"
ERROR snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py::test_pfc_pause_multi_lossless_prio[multidut_port_info1] - Failed: Processes "['analyze_logs--<MultiAsicSonicHost yy39top-lc4>']" failed with exit code "1"
ERROR snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py::test_pfc_pause_multi_lossless_prio_reboot[multidut_port_info1-fast] - Failed: Not all ports that are admin up on are operationally up
FAILED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio[multidut_port_info0-yy39top-lc4|3] - Failed: PFC pause frames should be transmitted and counted in TX PFC counters for priority 3
FAILED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio[multidut_port_info0-yy39top-lc4|4] - Failed: PFC pause frames should be transmitted and counted in TX PFC counters for priority 4
FAILED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py::test_pfc_pause_multi_lossless_prio[multidut_port_info0] - Failed: PFC pause frames should be transmitted and counted in TX PFC counters for priority 3
FAILED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio_reboot[multidut_port_info0-cold-yy39top-lc4|3] - Failed: PFC pause frames should be transmitted and counted in TX PFC counters for priority 3
FAILED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py::test_pfc_pause_single_lossless_prio_reboot[multidut_port_info0-cold-yy39top-lc4|4] - Failed: PFC pause frames should be transmitted and counted in TX PFC counters for priority 4
FAILED snappi_tests/multidut/pfc/test_multidut_pfc_pause_lossless_with_snappi.py::test_pfc_pause_multi_lossless_prio_reboot[multidut_port_info0-cold] - Failed: PFC pause frames should be transmitted and counted in TX PFC counters for priority 3
======================================================================================== 6 failed, 6 passed, 12 skipped, 16 warnings, 5 errors in 11482.18s (3:11:22) ========================================================================================
sonic@ixia-sonic-mgmt-whitebox:/data/tests$

```

Reply:
And after fixing the RP, we see full pass with test_multidut_snappi.py:

```
=========================================================================================================================== PASSES ===========================================================================================================================
______________________________________________________________________________________________________________ test_snappi[multidut_port_info0] ______________________________________________________________________________________________________________
______________________________________________________________________________________________________________ test_snappi[multidut_port_info1] ______________________________________________________________________________________________________________
----------------------------------------------------------------------------- generated xml file: /run_logs/ixia/fixturize-reboot/2024-10-30-02-19-21/tr_2024-10-30-02-19-21.xml -----------------------------------------------------------------------------
INFO:root:Can not get Allure report URL. Please check logs
------------------------------------------------------------------------------------------------------------------- live log sessionfinish -------------------------------------------------------------------------------------------------------------------
02:25:43 __init__.pytest_terminal_summary         L0067 INFO   | Can not get Allure report URL. Please check logs
================================================================================================================== short test summary info ===================================================================================================================
PASSED snappi_tests/test_multidut_snappi.py::test_snappi[multidut_port_info0]
PASSED snappi_tests/test_multidut_snappi.py::test_snappi[multidut_port_info1]
ERROR snappi_tests/test_multidut_snappi.py::test_snappi[multidut_port_info0] - Failed: Processes "['analyze_logs--<MultiAsicSonicHost yy39top-lc4>']" failed with exit code "1"
===================================================================================================== 2 passed, 5 warnings, 1 error in 379.58s (0:06:19) =====================================================================================================
sonic@ixia-sonic-mgmt-whitebox:/data/tests$
```